run_project_tests.py

```diff
@@ -85,7 +93,16 @@ def __exit__(self, _type, value, traceback):
         # holding files open.
         mesonlib.windows_proof_rmtree(self.dir)
 
+compile_commands = None  # type: str
+clean_commands = None  # type: str
+install_commands = None  # type: str
+uninstall_commands = None  # type: str
+backend_flags = []  # type: typing.List[str]
+executor = None  # type: ignore
+logfile = None  # type: typing.TextIO
+backend = None  # type: ignore
 failing_logs = []
+futures = []  # type: ignore
```

Comment:
This should at the very least be a `typing.List[typing.Any]`, but I think it holds `concurrent.Future`s, right?